src/app/modules/main/wallet_section/collectibles/models/collectibles_model.nim

```diff
@@ -100,9 +130,25 @@ QtObject:
       let traits = newTraitModel()
       traits.setItems(item.getStats())
       result = newQVariant(traits)
+    of CollectibleRole.CollectionName:
+      result = newQVariant(item.getCollectionName())
+    of CollectibleRole.CollectionSlug:
+      result = newQVariant(item.getCollectionSlug())
+    of CollectibleRole.CollectionImageUrl:
+      result = newQVariant(item.getCollectionImageUrl())
+    of CollectibleRole.IsLoading:
+      result = newQVariant(item.getIsLoading())
 
   proc getItem*(self: Model, index: int): Item =
     return self.items[index]
+  
+  proc setIsFetching*(self: Model, value: bool) =
+    self.isFetching = value
```

Comment:
Add an early return if the values are the same

Reply:
Done